pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -387,9 +387,13 @@ func (e *aksOperatorController) deployAKSOperator() error {
 	return e.DeployOperator(aksOperator, aksOperatorTemplate, aksShortName)
 }
 
-func (e *aksOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, bool, error) {
+func (e *aksOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, *bool, error) {
```

Comment:
So if I understand this method correctly, for `requiresTunnel` we only return true or false if everything works and we are certain of the cluster case. Nil means error.

Reply:
Yes, that's the overall gist.